### PR TITLE
Don't retry by default when the stream is exhausted

### PR DESCRIPTION
**Pull request by shsms**
This change the default behaviour of `GrpcStreamBroadcaster` to not reconnect when the server closes the stream.  This can be overridden by setting the `retry_on_exhausted_stream` parameter to `True`.

### CODE REVIEW

**Review by llucax — COMMENTED** [review]
LGTM. I think it makes sense in general, and will be useful when the microgrid API start streaming set power results (instead of using just one reply), but I wonder if you have any other use case where this is needed now.

Not approving because the CI is failing (and needs release notes), so I will probably need to approve again anyways.

**Comment by shsms** [comment]
> but I wonder if you have any other use case where this is needed now.

The electricity trading API makes time limited streaming requests and at the end, the streamers need to exit, but they keep trying to reconnect and spam the logs.

I've updated the tests and added release notes.

File: tests/streaming/test_grpc_stream_broadcaster.py
Comment (llucax) [inline tests/streaming/test_grpc_stream_broadcaster.py:1]: I was really confused by the parametrization with only one value. Given that the tests with `retry_on_exhausted_stream` with `True` and `False` are almost identical[^1], wouldn't it make more sense to just take the argument in the test too and check differently for each value?

[^1]: Diff:
    ```diff
    @@ -15,7 +15,10 @@
             receiver_ready_event.set()
             async for item in receiver:
                 items.append(item)
    -    no_retry.next_interval.assert_called_once_with()
    +    assert (
    +        no_retry.next_interval.call_count == 0
    +    ), "next_interval should not be called when streaming is successful"
    +
         assert items == [
             "transformed_0",
             "transformed_1",
    @@ -26,8 +29,7 @@
         assert caplog.record_tuples == [
             (
                 "frequenz.client.base.streaming",
    -            logging.ERROR,
    -            "test_helper: connection ended, retry limit exceeded (mock progress), "
    -            "giving up. Stream exhausted.",
    +            logging.INFO,
    +            "test_helper: connection closed, stream exhausted",
             )
         ]
    ```

**Review by llucax — APPROVED** [review]
Only one thing about the tests that really got me thinking for a while, so it might be worth changing. I will approve because I don't consider this a blocker, but I disabled auto-merge in case you want to fix it.

File: tests/streaming/test_grpc_stream_broadcaster.py
Comment (shsms) [reply tests/streaming/test_grpc_stream_broadcaster.py:1]: I can only think of parameterising the cases as a way to make it look nice, but given that there are only two options, I'm not sure it is worth it.

**Pull request merged**